test/models/stacks_test.rb

```diff
@@ -262,6 +271,17 @@ def setup
       assert @stack.active_task?
     end
 
+    test ".run_deploy_in_foreground triggers a deploy" do
+      stack = Stack.create!(repo_owner: 'foo', repo_name: 'bar', environment: 'production')
+      commit = shipit_commits(:first)
+      stack.commits << commit
+
+      Stack.any_instance.stubs(:trigger_deploy).with(anything, is_a(AnonymousUser), env: [], force: true, skip_queue: true)
```

Comment:
I wasn't able to replace the `anything` with the `is_a?(Commit)` object as it wasn`t the same class (Commit vs ActiveRecord_Relation).
I tried `.with { |actual| actual }` but the block argument didn't include the right number of passed arguments. 
https://www.rubydoc.info/github/floehopper/mocha/Mocha/Expectation#with-instance_method

Reply:
Oh was this supposed to be an `expect` vs `stub` ? 

I'm fine with the `with` call, though really the only field we really care about is that it's using `skip_queue: true` right? If so I would be fine with putting `anything` for the rest of the fields so that we can really narrow the focus on this test (also becomes more resilient to other minor changes to this method's signature)

Reply:
This method kind of functions as a procedure, which makes it hard to test. The `with` expectation works as an expectation in this case. Suggestions on a better way to test this are welcome :)

Reply:
I believe that with the `stubs.with`, it'll only fail if the method was called with the wrong arguments. Vs. if you go with `expects.with`, it'll fail if the method was not called at all, or if it was called with the wrong arguments